packages/web/src/common/store/upload/sagaHelpers.ts

```diff
@@ -127,17 +127,23 @@ export function* addPremiumMetadata<T extends TrackMetadata>(track: T) {
   )
   if (!isUsdcPurchaseEnabled) return track
 
+  // download_conditions could be set separately from stream_conditions, so we check for them first
+  if (isContentUSDCPurchaseGated(track.download_conditions)) {
```

Comment:
curious why this bug didn't materialize sooner. your last PR that touched this was merged on April 5: https://github.com/AudiusProject/audius-protocol/pull/7982/files#diff-81d88197170465ad31b36ee1ea05c0cd103e04c0e7587b12d79f5287970d0612
@DejayJD do you have any ideas?

Reply:
I'm also curious and would like to investigate. My guess is maybe the more recent marshal changes? It is definitely possible that we didn't catch it though; I'm not certain that we were specifically testing editing of "tracks"

Reply:
normally the download conditions would always be present.
I did test editing of tracks. still possible i missed this but i thought that download conditions would be there normally